File: tse/City.cs
using System;

class City<T>
{
    private string name;
    private T x;
    private T y;

    public City(string name, T x, T y)
    {
        this.name = name;
        this.x = x;
        this.y = y;
    }

    public string GetName()
    {
        return this.name;
    }

    public T GetX()
    {
        return this.x;
    }

    public T GetY()
    {
        return this.y;
    }

    public double DistanceTo(City<T> city) 
    {   
        
        double xDistance = Math.Abs(Convert.ToDouble(GetX()) - Convert.ToDouble(city.GetX()));
        double yDistance = Math.Abs(Convert.ToDouble(GetY()) - Convert.ToDouble(city.GetY()));
        double distance = Math.Sqrt((xDistance * xDistance) + (yDistance * yDistance));
        return distance;
    }

    public override string ToString()
    {
        return GetName();
    }
}

File: versionJava/City.java
class City{
    private String name;
    private double x;
    private double y;

    public City(String name, double x, double y){
        this.name = name;
        this.x = x;
        this.y = y;
    }

    public String getName(){
        return this.name;
    }

    public double getX(){
        return this.x;
    }

    public double getY(){
        return this.y;
    }
    public double distanceTo(City city){
        double xDistance = Math.abs(getX() - city.getX());
        double yDistance = Math.abs(getY() - city.getY());
        double distance = Math.sqrt( (xDistance*xDistance) + (yDistance*yDistance) );
        return distance;
    }
    @Override
    public String toString() {
        return getName();
    }
}
